// Copyright 2025 Ian Lewis
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

// Provider represents a currency exchange provider.
message Provider {
    // Code is a URL-fragment code for the provider.
    string code = 1; 
    
    // name is the human readable name for the provider.
    string name = 2;

    // supported_base_currencies is the list of provider's supported base currencies.
    repeated string supported_base_currencies = 3;

    // supported_quote_currencies is the list of provider's supported quote currencies.
    repeated string supported_quote_currencies = 4;
}

// ProviderList represents a lits of Provider objects.
message ProviderList {
    repeated Provider providers = 1;    
}
